### PR TITLE
[Inference] Support `stop` parameter in `text-generation` instead of `stop_sequences`

### PR DESCRIPTION
**Pull request by Wauplin**
Fix https://github.com/huggingface/huggingface_hub/issues/2471 cc @sadra-barikbin.

This PR deprecates `stop_sequences` in favor of the `stop` parameter for the text_generation task.

**Context:** in both TGI and the `text_generation` [specs](https://github.com/huggingface/huggingface.js/blob/1de39598a231afab805e252d2b27e0ec56a1897a/packages/tasks/src/tasks/text-generation/spec/input.json#L94) the `stop` parameter is provide stop tokens to the model. However historically `transformers` was using the `stop_sequences` parameter which had been propagated to Inference API and `InferenceClient`. Since we are now TGI-first (i.e. even `transformers` models are served with TGI), let's just expose `stop`.

```py
>>> from huggingface_hub import InferenceClient
>>> InferenceClient("gpt2").text_generation("The capital of France is", stop=["Republic"])
 the capital of the French Republic
```

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Makes sense! cc @gante @ArthurZucker maybe at some point we'll want to align `stop_sequences` to `stop` if that's what's used everywhere else

File: src/huggingface_hub/inference/_client.py [hunk 1]
```diff
@@ -1655,7 +1655,8 @@ def text_generation(  # type: ignore
         repetition_penalty: Optional[float] = None,
         return_full_text: Optional[bool] = False,  # Manual default value
         seed: Optional[int] = None,
-        stop_sequences: Optional[List[str]] = None,  # Same as `stop`
+        stop: Optional[List[str]] = None,
+        stop_sequences: Optional[List[str]] = None,  # Deprecated, use `stop` instead
```
Comment (osanseviero) [inline src/huggingface_hub/inference/_client.py:1659]: Note that `transformers` now uses `stop_strings` but as you mention, `transformers` will be deployed with TGI so not big deal

**Review by osanseviero — APPROVED** [review]
Thanks!

**Comment by Wauplin** [comment]
Thanks both for the reviews!

File: src/huggingface_hub/inference/_client.py (hunk 1 above)
Comment (osanseviero) [reply src/huggingface_hub/inference/_client.py:1659]: Actually this will be an issue with IE deployment if not using TGI right?

Comment (Wauplin) [reply src/huggingface_hub/inference/_client.py:1659]: Yes and no, it's harder than that as you found out (private [gdoc](https://docs.google.com/document/d/1asPoVRbR7-H3Bd16E3A0p4sRcSZ11bQbf36BrICrLVo/edit?usp=sharing))

**Pull request merged**